app/models/building_operator.rb

```diff
@@ -85,6 +86,24 @@ def questions_by_building_type(building_type_id)
   end
 
   def questions_by_building(building_id)
+    # Returns
+    # questions: array of question objects that are accessible to read
+    questions = []
+    d = delegations.includes(
```

Comment:
Same here. Use delegations.join and I think you can apply conditions on delegation and delegation.answer before iteration.